command/services/config_test.go

```diff
@@ -15,10 +17,18 @@ import (
 // We depend on this behavior for ServiesFromFiles so we want to fail
 // tests if that ever changes.
 func TestDevModeHasNoServices(t *testing.T) {
+	hostname, err := os.Hostname()
+	require.NoError(t, err)
 	devMode := true
-	result, err := config.Load(config.LoadOpts{DevMode: &devMode})
+	opts := config.LoadOpts{DevMode: &devMode}
+	result, err := config.Load(opts)
 	require.NoError(t, err)
-	require.Len(t, result.Warnings, 0)
+	if !strings.Contains(hostname, ".") {
```

Comment:
Generally I try to avoid any conditional assertions in tests, especially when they are for handling differences in the test environment.

I think you could handle it this way instead:

```go
opts := config.LoadOpts{
    DevMode: &devMode
    HCL: []string{`node_name = "dummy"`},
}

That should do about the same thing as the other tests.
```

Reply:
Should I add a test to validate the current behaviour when a node name is invalid and that we get the right warning?

Reply:
fixed the test and added an extra test for invalid `node_name` use case

Reply:
I think we should already have test coverage for the warnings.  I wouldn't worry about adding a new one since this PR doesn't change that behaviour.

Reply:
Ah, I see you added one, that looks good too!